Tools/LyTestTools/ly_test_tools/o3de/editor_test_utils.py

```diff
@@ -0,0 +1,144 @@
+    """
+    return the full path of the test module using always '.py' extension
+    :param testcase_module: The testcase python module being tested
+    :return str: The full path to the testcase module
+    """
+    return os.path.splitext(testcase_module.__file__)[0] + ".py"
+
+def get_module_filename(testcase_module):
+    # type: (Module) -> str
+    """
+    return The filename of the module without path
+    Note: This is differs from module.__name__ in the essence of not having the package directory.
+    for example: /mylibrary/myfile.py will be "myfile" instead of "mylibrary.myfile"
+    :param testcase_module: The testcase python module being tested
+    :return str: Filename of the module
+    """
+    return os.path.splitext(os.path.basename(testcase_module.__file__))[0]
+
+def retrieve_log_path(run_id : int, workspace):
+    """
+    return the log/ project path for this test run.
+    :param run_id: editor id that will be used for differentiating paths
+    :param workspace: Workspace fixture
+    :return str: The full path to the given editor the log/ path
+    """
+    return os.path.join(workspace.paths.project(), "user", f"log_test_{run_id}")
+
+def retrieve_crash_output(run_id : int, workspace, timeout : float):
+    """
+    returns the crash output string for the given test run.
+    :param run_id: editor id that will be used for differentiating paths
+    :param workspace: Workspace fixture
+    :timeout: Maximum time (seconds) to wait for crash output file to appear
+    :return str: The contents of the editor crash file (error.log)
+    """
+    crash_info = "-- No crash log available --"
+    crash_log = os.path.join(retrieve_log_path(run_id, workspace), 'error.log')
+    try:
+        waiter.wait_for(lambda: os.path.exists(crash_log), timeout=timeout)
+    except AssertionError:                    
+        pass
+        
+    # Even if the path didn't exist, we are interested on the exact reason why it couldn't be read
+    try:
+        with open(crash_log) as f:
+            crash_info = f.read()
+    except Exception as ex:
+        crash_info += f"\n{str(ex)}"
+    return crash_info
+
+def cycle_crash_report(run_id : int, workspace):
+    """
+    Attempts to rename error.log and error.dmp(crash files) into new names with the timestamp on it.
+    :param run_id: editor id that will be used for differentiating paths
+    :param workspace: Workspace fixture
+    """
+    log_path = retrieve_log_path(run_id, workspace)
+    files_to_cycle = ['error.log', 'error.dmp']
+    for filename in files_to_cycle:
+        filepath = os.path.join(log_path, filename)
+        name, ext = os.path.splitext(filename)
+        if os.path.exists(filepath):
+            try:
+                modTimesinceEpoc = os.path.getmtime(filepath)
+                modStr = time.strftime('%Y_%m_%d_%H_%M_%S', time.localtime(modTimesinceEpoc))
+                new_filepath = os.path.join(log_path, f'{name}_{modStr}{ext}')
```

Comment:
I'm concerned this will exceed the max character limit constantly. Is there a way to reduce the filename?

Reply:
editor_XXXX_XX_XX_XX_XX_XX.log is the maximum filename length here, there are files that are longer, so shouldn't be a concern